apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx

```diff
@@ -57,33 +59,102 @@ export default function CurriculumQuickAssign({
   }, [participantType]);
 
   useEffect(() => {
-    if (!courseOfferings) return;
+    // Filter the offerings based on the filters provided
+    const filterOfferings = data => {
+      const languageFilter = courseFilters?.language;
+
+      if (languageFilter) {
+        // Crawl data and remove any courses / versions that are not available
+        // in the requested language.
+        for (const levelInfo of Object.values(data)) {
+          // For each level (elementary, middle, high, hoc, etc), go through the categories
+          for (const [categoryKey, categoryInfo] of Object.entries(levelInfo)) {
+            // Here, we are listing the category (Course, Module, etc)
+            // Now for each course, we go through the versions and filter out only ones
+            // matching the requested language (CS Fundamentals, Express, etc)
+            if (Array.isArray(categoryInfo)) {
+              for (const courseInfo of categoryInfo) {
+                // These will be the course info blocks which are a tuple of the id and then metadata.
+                courseInfo.course_versions = courseInfo.course_versions.filter(
+                  ([_, versionInfo]) =>
+                    versionInfo.locale_codes.includes(languageFilter)
+                );
```

Comment:
Since this same filter appears twice, it might be worth pulling it into its own function.